addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -190,7 +192,10 @@ function insert(editor, data, isText = true) {
     currentNode = lastChildNode || currentNode;
     selection.removeAllRanges();
     const newRange = new Range();
-    const lastPosition = rightPos(currentNode);
+    let lastPosition = rightPos(currentNode);
+    if (lastPosition[0] === editor.editable) {
+        lastPosition = getDeepestPosition(...lastPosition);
+    }
```

Comment:
I'm not sure I understand why this is needed. I'm sure there is a good reason, but the commit message wasn't really helping as it only speaks of "an extra issue with the positioning of the selection after the insertion". I think this warrants a comment in the code. In fact I'd also split it in two different commits rather than passing this fix kind of "under the rug" of the "main" fix, but it's not that big of a deal if you like it in one commit.

Reply:
You're right. This is to handle the issue you mentioned here: https://github.com/odoo-dev/odoo/commit/d434d502238ab58759e622204754e061e08a171e#r91535151

I'll split into two commits.

Reply:
Done. But I think I remember there were talks of an option for the cursor to be allowed directly in the root (like if the root is inline I guess). Was that done? If so, I have to include it in the if and we should check if it's taken into account in `getDeepestPosition`.